clang/lib/Tooling/DependencyScanning/DependencyScanningWorker.cpp

```diff
@@ -142,16 +142,17 @@ class DependencyScanningAction : public tooling::ToolAction {
 public:
   DependencyScanningAction(
       StringRef WorkingDirectory, DependencyConsumer &Consumer,
+      const CASOptions &CASOpts,
       llvm::IntrusiveRefCntPtr<DependencyScanningWorkerFilesystem> DepFS,
       llvm::IntrusiveRefCntPtr<DependencyScanningCASFilesystem> DepCASFS,
       ExcludedPreprocessorDirectiveSkipMapping *PPSkipMappings,
       bool OverrideCASTokenCache, ScanningOutputFormat Format,
       bool OptimizeArgs, bool EmitDependencyFile,
       llvm::Optional<StringRef> ModuleName = None)
       : WorkingDirectory(WorkingDirectory), Consumer(Consumer),
-        DepFS(std::move(DepFS)), DepCASFS(std::move(DepCASFS)),
-        PPSkipMappings(PPSkipMappings), Format(Format),
-        OverrideCASTokenCache(OverrideCASTokenCache),
+        CASOpts(CASOpts), DepFS(std::move(DepFS)),
```

Comment:
Seems like `std::move(CASOpts)` would be better.

Reply:
Do you mean that I should change `DependencyScanningAction.CASOpts` to be a copy (`CASOptions CASOpts;`) instead of a reference (`const CASOptions &CASOpts;`)?

Note that `DependencyScanningAction` is not self-contained, there are these fields:
```
  DependencyConsumer &Consumer;
  ExcludedPreprocessorDirectiveSkipMapping *PPSkipMappings;
```
which indicates that `DependencyScanningAction` should not outlive `DependencyScanningWorker` or its caller, so there's no gain to avoid the reference, AFAICT.

Reply:
@dexonsmith I'll go ahead and merge but let me know if you think I should make a change here and I'll follow-up.

Reply:
I must have misread this, and thought this was a constructor for one of the fields where it *was* a copy. 141d745a476f8093fad48bf7e9809d3fe6dc8289 LGTM!